app/javascript/src/components/editor/Editor.svelte

```diff
@@ -91,37 +92,42 @@
       const useParameterObject = $settings["autocomplete-parameter-objects"] && params.args_length >= $settings["autocomplete-min-parameter-size"]
       const useNewlines = params.args_length >= $settings["autocomplete-min-parameter-newlines"]
 
-      const apply = v.args.map(a => {
+      // Generate Apply map to be used for autocomplete and other bits
+      let apply = v.args.map(a => {
         const name = toCapitalize(a.name?.toString().toLowerCase())
         let defaultValue = a.default?.toString().toLowerCase().replaceAll(",", "")
-        
+
         if (lowercaseDefaults.includes(defaultValue.toLowerCase())) defaultValue = defaults[toCapitalize(defaultValue)]
         else defaultValue = toCapitalize(defaultValue)
 
-        if (useParameterObject) return `${ useNewlines ? "\n\t" : "" } ${ name }: ${ defaultValue }`
-        
-        return defaultValue
+        return [name, defaultValue]
       })
 
+      // Set completion map keys and default
       params.parameter_keys = detail
-      params.parameter_defaults = apply
+      params.parameter_defaults = apply.map(([_, defaultValue]) => defaultValue)
+
+      // If useParameterObject is enabled transform the apply to include the parameter name.
+      // It's important this happens after setting the parameter_defaults param, as that uses
+      // a different format and we don't want it to use the parameter object format.
+      if (useParameterObject) apply = apply.map(([name, defaultValue]) => ([name, `${ useNewlines ? "\n\t" : "" } ${ name }: ${ defaultValue }`]))
 
+      const applyValues = apply.map(([_, defaultValue]) => defaultValue)
```

Comment:
We can avoid modifying `apply`, since it isn't used afterward 😁

```suggestion
      const applyValues = apply.map(([name, defaultValue]) => {
        // If useParameterObject is enabled transform the apply to include the parameter name.
        // It's important this happens after setting the parameter_defaults param, as that uses
        // a different format and we don't want it to use the parameter object format.
        if (useParameterObject) return `${ useNewlines ? "\n\t" : "" } ${ name }: ${ defaultValue }`

        return defaultValue
      })
```

Reply:
Ah sure, that seems like a better solution